--- a/llvm/lib/Support/TrieRawHashMap.cpp
+++ b/llvm/lib/Support/TrieRawHashMap.cpp
@@ -108,7 +108,7 @@
   unsigned NumBits = 0;
   unsigned Size = 0;
   friend class llvm::ThreadSafeTrieRawHashMapBase;
-  friend class TrailingObjects;
+  friend TrailingObjects;
 
 public:
   /// Linked list for ownership of tries. The pointer is owned by TrieSubtrie.
@@ -201,7 +201,7 @@
   ThreadSafeAllocator<BumpPtrAllocator> ContentAlloc;
 
 private:
-  friend class TrailingObjects;
+  friend TrailingObjects;
 
   ImplType(size_t StartBit, size_t NumBits) {
     ::new (getRoot()) TrieSubtrie(StartBit, NumBits);
